### PR TITLE
QOL Fixes

### PR DESCRIPTION
**Pull request by pierssinclairssw**

### CODE REVIEW

**Review by ozairashfaqueSSW — APPROVED** [review]

File: rules/relational-database-design/rule.md [hunk 1]
```diff
@@ -195,25 +196,38 @@ So to fix it, Book needs to become a separate table.
 
 ERDs are a way of visualizing the relationships between different entities and their cardinality. 
 
+`youtube: https://www.youtube.com/watch?v=6uwuNRUUimY`
```
Comment (ncn-ssw) [inline rules/relational-database-design/rule.md:199]: Recommend line break after the YouTube video to ensure that the Figure caption is on the next line.

File: rules/relational-database-design/rule.md [hunk 2]
```diff
@@ -26,13 +26,14 @@ These problems can be mitigated using Entity-Relationship Diagrams (ERDs) and Da
...
-•	First Normal Form (1NF)
-•	Second Normal Form (2NF)
-•	Third Normal Form (3NF)
+- First Normal Form (1NF)
+- Second Normal Form (2NF)
+- Third Normal Form (3NF)
 
 There are [further normal forms](https://en.wikipedia.org/wiki/Database_normalization), and the theory constantly evolves, but these 3 are the bare minimum required for database design.
 
 `youtube: https://youtu.be/J-drts33N8g`
```
Comment (ncn-ssw) [inline rules/relational-database-design/rule.md:35]: Recommend line break after the YouTube video to ensure that the Figure caption is on the next line.

File: rules/relational-database-design/rule.md [hunk 3]
```diff
@@ -195,25 +196,38 @@ So to fix it, Book needs to become a separate table.
...
+* **One-to-One**: Each instance of entity A can be associated with one instance of entity B, and vice versa.
+  * e.g. each employee in a company has one employee ID, and each employee ID is associated with one employee.
+* **One-to-Many**: Each instance of entity A can be associated with multiple instances of entity B, but each instance of entity B is associated with only one instance of entity A.
+  * e.g. one author can write many books, but each book is written by one author.
+* **Many-to-Many**: Each instance of entity A can be associated with multiple instances of entity B, and vice versa.
+  * e.g. a student can enrol in multiple courses, and multiple students can take a course.
 
-To effectively handle many-to-many relationships in a relational database, we use an associative entity (also known as a joining table) to convert the many-to-many relationship into two one-to-many relationships.
+### Optionality
+In addition to the above relationship types, each side of the relationship may be optional. Let's examine the case of an airline company which tracks Pilots and Completed Flights
```
Comment (ncn-ssw) [inline rules/relational-database-design/rule.md:217]: Recommend adding a full stop at the end of the sentence.

File: rules/relational-database-design/rule.md [hunk 4]
```diff
@@ -195,25 +196,38 @@ So to fix it, Book needs to become a separate table.
...
+- 1 Flight must always have at least one Pilot
+- 1 Pilot can have 0, 1 or many Flights (they may have 0 if they have been recently hired)
 
-### Practical Example - Student Courses
+So in this case, the Pilot having Flights is optional
```
Comment (ncn-ssw) [inline rules/relational-database-design/rule.md:221]: Recommend adding a full stop at the end of the sentence.

**Review by ncn-ssw — COMMENTED** [review]
Only minor punctuation issues noted.

**Pull request merged**